### ISSUE TITLE
test(symbols): prove one source-inventory-to-provider vertical and retire split authorities

### ISSUE THREAD

**Issue by EffortlessSteven** [EffortlessMetrics/perl-lsp-swarm#9269]
Programme/controller: follow-up controller issue in this train
Boundary parent: #8271
Source inventory: #8580 / #8611 / #8648
Declaration consumer convergence: #4047
Semantic contribution/projection: #7306 / #8683 / #8713
Reference proof/cutover: #7557 / #8083 / #8168
Workspace name view: #8262 / #8756
LSP projection policy: #8794
Relation cleanup: #9267
Rename exactness: #8511 / #7482
Published API closeout: #9268
Provider outcome proof: #8104
Installed/composed project journeys: #4245

Audit basis: `main@62c1d0a664c25d3a4c5741040281d014836d8bfe`.

## Priority

P1 composed proof and split-authority retirement. One PR after the implementation leaves in this train are merged or explicitly bounded.

## Problem

Component migrations can all be locally correct while the live system still contains a hidden duplicate producer, stale materialization, compatibility rescan, name-only fallback, or provider projection that bypasses the canonical path.

A source inventory unit test does not prove semantic projection. Semantic parity does not prove the workspace/materialized view. A correct internal result does not prove the exact wire response. A clean public API does not prove production stopped calling an old path.

The train needs one final executable vertical whose job is to challenge the integrated authority chain and prove the old split authorities are actually unreachable.

## Objective

Prove one candidate-bound vertical:

```text
accepted source/parser snapshot
→ one SourceSymbolInventory traversal
→ canonical declaration/occurrence/boundary contributions
→ semantic entity/binding/relation resolution
→ ProjectModel/workspace/query materialization
→ document symbols
→ workspace symbols
→ references
→ exact rename for an admitted cohort or typed refusal
→ exact external responses
```

The proof must bind every stage to the same source/document/workspace/root/configuration generations and distinguish exact, partial, bounded, stale, dynamic, unsupported, and instrument-failed states.

## Required authority counters

For each admitted journey record and assert equivalent counters:

```text
canonical source inventory producers        = 1
canonical AST source traversals              = 1 per source build
legacy declaration producer traversals       = 0
legacy occurrence/reference traversals        = 0
provider-local semantic AST walks            = 0
provider/workspace source rescans             = 0
production perl-symbol SymbolIndex users      = 0
lower LSP kind mapping calls                  = 0
lower workspace query-policy calls            = 0
synthetic/sentinel exact semantic facts       = 0
coarse-kind/name edit-authority uses          = 0
stale publication accepted                    = 0
```

Instrumentation failure is `not_proven`, not zero.

### Identity collisions

- same spelling in two lexical scopes;
- same subroutine/method/constant spelling in one package;
- same spelling in two packages and two roots;
- repeated declaration/occurrence byte offsets in distinct sources;
- reopened package declarations;
- generated/source-backed and generated-no-source members;
- import/re-export aliases preserving target kind and relation identity.

### Source geometry and boundaries

- exact declaration name spans;
- exact variable/call/method/coderef/typeglob anchors;
- Unicode/non-BMP and LF/CRLF/bare-CR rows;
- comments, POD, strings, regexes, heredocs, formats, `__DATA__` lookalikes;
- dynamic typeglob/method selector/AUTOLOAD/symbolic boundary;
- recovered source with exact and unavailable anchors.

### Lifecycle/currentness

Exercise:

- initial open/index;
- edit adding/removing one declaration;
- edit adding/removing one reference without changing the definition file;
- source-identical later generation;
- declaration/reference rename;
- close/reopen with same URI/version/bytes;
- file delete/re-add;
- file rename;
- workspace root removal/addition;
- configuration/readiness transition where relevant;
- held old inventory/semantic/index/provider work released after newer state.

Old contributions and responses must become unavailable or stale before replacements become current.

### Document symbols

- consume canonical declaration/container identity;
- exact selection ranges come from source-owned name spans;
- LSP `SymbolKind` comes from #8794 typed projection;
- source order and duplicate names remain deterministic.

### Workspace symbols

- full and accelerated paths agree under #8262;
- candidate name view is generation-owned under #8756;
- partial/unready zero observed matches cannot become exact empty;
- same spelling in multiple sources/roots remains distinct.

### References

- provider consumes #8168 identity-preserving semantic query path;
- exact empty and unavailable facts remain distinct;
- dynamic/ambiguous boundaries suppress exactness rather than disappear;
- no old AST/source scan contributes final membership.

### Rename

Use one already-admitted exact cohort from #7482. Assert:

```text
authorized semantic edit set == emitted edit set
```

- #8511 prevents coarse kind affinity from widening membership;
- comments/POD/strings/regex/heredocs/DATA same-text controls remain untouched;
- unresolved/dynamic/partial/stale cases return typed refusal with zero partial edits.

## External response binding

Use #8104/provider-outcome machinery where applicable. For each provider row bind:

```text
request identity
accepted source/document/workspace/root/config generations
internal provider decision
canonical producer/query identity
external response/edit identity
independent expected result
```

A correct receipt without its matching response, or a response from another generation/candidate, is not proof.

## Package/API recurrence checks

Consume #9268 checks and additionally prove production has no imports/calls to retired compatibility surfaces. A compile-only compatibility wrapper may remain only if its issue disposition explicitly permits it and counters prove it is not on the live path.

## Mutation/falsifier controls

The focused gate must fail when a mutation:

1. restores an independent declaration AST walk;
2. restores an independent reference AST walk;
3. drops a dynamic boundary and reports exact empty;
4. fabricates a `FileId(0)`/sentinel/synthetic source endpoint;
5. uses first candidate or name-only target selection;
6. restores `kind_compatible`-style edit membership;
7. restores mutable production `perl_symbol::SymbolIndex` use;
8. changes full/accelerated workspace symbol results;
9. restores raw lower LSP symbol mapping;
10. publishes old-generation contributions after edit/delete/root removal;
11. matches provider receipt to the wrong response;
12. returns a partial rename edit set;
13. lets one passing provider hide failure in another plane;
14. removes instrumentation and treats missing counts as zero.

## Generated receipt

Emit a deterministic candidate-bound `symbol_authority_vertical.v1` or equivalent containing:

```text
candidate/build/profile identity
source/parser/inventory schema identities
authority-ledger digest
semantic/project/query schema identities
provider/request/client profile identities
fixture/source/root/generation identities
authority counters
per-plane exact/bounded/refused/not-proven result
external response/edit digest
stale/dynamic/boundary observations
mutation/instrument status
reproduce command
```

No aggregate percentage substitutes for row-level results.

## Acceptance

- [ ] One source inventory is the only source declaration/occurrence producer for the admitted vertical.
- [ ] Semantic/project/query layers enrich/materialize without source rewalks or identity reconstruction.
- [ ] Workspace name acceleration is correctness-neutral and generation-owned.
- [ ] LSP presentation policy is above `perl-symbol`.
- [ ] Import/export relation projections preserve intrinsic entity identity.
- [ ] References preserve exact source/semantic/project identity and completeness.
- [ ] One exact rename cohort proves semantic authorized-set equality and safe refusal boundaries.
- [ ] Exact external responses bind to matching internal decisions/current generations.
- [ ] Lifecycle transitions remove stale contributions and reject late work.
- [ ] Authority counters and mutation controls prove split production authorities are unreachable.
- [ ] #9268 public/package recurrence checks are green.
- [ ] Missing instrumentation/evidence is `not_proven`, never pass.

## Verification

Run the focused commands from every consumed leaf plus:

```bash
cargo test -p perl-symbol --all-targets --locked
cargo test -p perl-semantic-analyzer --all-targets --locked
cargo test -p perl-workspace --all-targets --locked
cargo test -p perl-lsp-rs-core --all-targets --locked
cargo test -p perl-lsp-rs --all-targets --locked
cargo xtask check-architecture
cargo xtask semantic-shadow-compare --check
cargo xtask semantic-scorecard --check
cargo fmt --all -- --check
cargo clippy --workspace --all-targets -- -D warnings
git diff --check
```

## Codex boundary

Do not fix discovered semantic/provider defects by weakening this proof or adding compatibility fallbacks. File/narrow the exact defect if it belongs outside the train; fix train-owned recurrence directly in its owner. This PR owns composition, falsification, receipts, and old-authority retirement proof—not new feature breadth.

## Non-goals

- No new semantic engine.
- No all-provider/all-Perl support claim.
- No parser algorithm work.
- No release/tag/publication mutation.
- No aggregate quality score.

**Comment by EffortlessSteven** [comment]
Workspace-symbol proof input refined: consume #10629 from controller #10632.

#10629 now proves the workspace-symbol portion of this integrated vertical across:

- canonical current base/open row population (#10641/#10624);
- one semantic match/dedup/order/product budget (#10642/#10626);
- one shared-service operation (#9015);
- one atomic LSP projection with zero silent row/collection loss and zero post-serialization root rematching (#10644);
- multi-root/currentness/lifecycle/exact-empty/bounds/response correlation.

For #9269 authority counters, treat #10629 as the current workspace-symbol receipt for:

```text
provider/workspace source rescans = 0
production perl-symbol SymbolIndex exact reads = 0
lower query-policy calls outside #8794 = 0
producer-local product caps = 0
silent projection row/collection loss = 0
stale publication accepted = 0
```

Do not recreate workspace-symbol expected results or an independent exact-process route inside #9269. #9269 still composes this with source inventory, document symbols, references, rename and public/package recurrence evidence.
